mailet/base/src/main/java/org/apache/mailet/base/GenericMailet.java

```diff
@@ -290,34 +290,18 @@ public void log(String message, Throwable t) {
     
     /**
      * Utility method: Checks if there are unallowed init parameters specified in the 
-     * configuration file against the String[] allowedInitParameters.
-     * @param allowedArray array of strings containing the allowed parameter names
+     * configuration file against the allowedInitParameters.
+     * @param allowed List of strings containing the allowed parameter names
      * @throws MessagingException if an unknown parameter name is found
      */
-    protected final void checkInitParameters(String[] allowedArray) throws MessagingException {
-        // if null then no check is requested
-        if (allowedArray == null) {
-            return;
-        }
-        
-        Collection<String> allowed = new HashSet<>();
-        Collection<String> bad = new ArrayList<>();
+    protected final void checkInitParameters(Set<String> allowed) throws MessagingException {
+        Set<String> bad = Streams.stream(getInitParameterNames())
+            .filter(not(allowed::contains))
+            .collect(Guavate.toImmutableSet());
 
-        Collections.addAll(allowed, allowedArray);
-        
-        Iterator<String> iterator = getInitParameterNames();
-        while (iterator.hasNext()) {
-            String parameter = iterator.next();
-            if (!allowed.contains(parameter)) {
-                bad.add(parameter);
-            }
-        }
-        
-        if (bad.size() > 0) {
-            throw new MessagingException("Unexpected init parameters found: " + Arrays.toString(bad.toArray()));
+        if (!bad.isEmpty()) {
```

Comment:
I think `bad` is not a good name, how about `reject` or something more clear?

Reply:
I didn't took care of that originally and to be honest, refactoring is an activity requiring to not care about everything if one want to settle to a PR.
I propose to not change that in this PR and you create a PR with this change if you care enough about it.

Reply:
ok.